app/interactors/change_history/update_interactor.rb

```diff
@@ -0,0 +1,35 @@
+class ChangeHistory::UpdateInteractor < ApplicationInteractor
```

Comment:
I've had a bit of a tinker and I think the best means to extract the logic of this would be to create a dedicated class in lib for this task. So say something like `lib/change_history_modifier.rb`.

I came up with something like this: 
```
class ChangeHistoryModifier
  def initialize(content_id, locale, user_email)
    @content_id = content_id
    @locale = locale || "en"
    @user = User.find_by!(email: user_email) if user_email
  end
  
  def add(note, timestamp)
  end
  
  def edit(id, note)
  end
  
  def delete(id)
    update_edition do |edition, updater|
      entry = edition.change_history.find { |item| item["id"] == id }
      raise "No change history entry with id #{id}" unless entry
      updater.assign(change_history: edition.change_history - entry)
    end
  end
  
private

  attr_reader :content_id, locale, :user

  def update_edition
    Edition.transaction do
      current_edition = Edition.lock.find_current(id: "#{content_id}:#{locale}")
      raise "Edition must be editable" unless edition.editable?
      
      updater = Versioning::RevisionUpdater.new(edition.revision, user)
      yield edition, updater
      
      raise "Expected an updated revision" unless updater.changed?
      
      EditDraftEditionService.call(edition, user, revision: updater.next_revision)
      edition.save!
      FailsafeDraftPreviewService.call(edition)
    end
  end
end
```

which I thought could also take a brunt of the testing load. It didn't feel quite right to use a service for this as there are a few different avenues, and probably don't want it too caught up in application logic since we want specific exceptions.

I'm imagining we may also need to generalise or extract some of the edition updating logic to also support this story: https://trello.com/c/jjrWQcWs/1319-implement-means-to-store-and-alter-alternative-format-organisation-in-content-publisher so we may want a dedicated class/module for edition updating by rake tasks that does this stuff: 

```
    Edition.transaction do
      current_edition = Edition.lock.find_current(id: "#{content_id}:#{locale}")
      raise "Edition must be editable" unless edition.editable?
      
      updater = Versioning::RevisionUpdater.new(edition.revision, user)
      yield edition, updater
      
      raise "Expected an updated revision" unless updater.changed?
      
      EditDraftEditionService.call(edition, user, revision: updater.next_revision)
      edition.save!
      FailsafeDraftPreviewService.call(edition)
```

Not too sure what to name it though to make it clear it's only for tasks 🤔